internal/test/cleanup/cleanup.go

```diff
@@ -101,10 +101,16 @@ func CleanUpCloudstackTestResources(ctx context.Context, clusterName string, dry
 	}
 	cmk := executableBuilder.BuildCmkExecutable(tmpWriter, execConfig.Profiles)
 	defer cmk.Close(ctx)
+
+	failedProfiles := []string{}
 	for _, profile := range execConfig.Profiles {
 		if err := cmk.CleanupVms(ctx, profile.Name, clusterName, dryRun); err != nil {
-			cmk.Close(ctx)
+			failedProfiles = append(failedProfiles, profile.Name)
```

Comment:
Sorry to keep going back and forth on this, it might be good to include the err message itself as well for debugging purposes